examples/moonshot-mainseries/00-prep-receptor.py

```diff
@@ -0,0 +1,422 @@
+        print('')
+        print('')
+        print(f'{complex_pdb_filename} : FAILURE')
+        print(warnings)
+        msg = f'No design units found for {complex_pdb_filename}\n'
+        msg += warnings
+        msg += '\n'
+        raise Exception(msg)
+
+    # Variants to generate
+    # (OEAtomMatchResidue, atom_name, formal_charge, implicit_hydrogen_count)
+    # OEAtomMatchResidue format is regex for (residue name, residue number, insertion code, chain ID, and fragment number)
+    variants = {
+        'His41(0)-Cys145(0)-His163(0)' : [ ('HIS:41:.*:.*:.*', 'ND1', 0, 0), ('HIS:41:.*:.*:.*', 'NE2', 1, 1), ('CYS:145:.*:.*:.*', 'SG', 0, 1), ('HIS:163:.*:.*:.*', 'NE2', 0, 0), ('HIS:163:.*:.*:.*', 'ND1', 1, 1) ],
+        'His41(0)-Cys145(0)-His163(+)' : [ ('HIS:41:.*:.*:.*', 'ND1', 0, 0), ('HIS:41:.*:.*:.*', 'NE2', 1, 1), ('CYS:145:.*:.*:.*', 'SG', 0, 1), ('HIS:163:.*:.*:.*', 'NE2', 1, 1), ('HIS:163:.*:.*:.*', 'ND1', 1, 1) ],
+        'His41(+)-Cys145(-)-His163(0)' : [ ('HIS:41:.*:.*:.*', 'ND1', 1, 1), ('HIS:41:.*:.*:.*', 'NE2', 1, 1), ('CYS:145:.*:.*:.*', 'SG', -1, 0), ('HIS:163:.*:.*:.*', 'NE2', 0, 0), ('HIS:163:.*:.*:.*', 'ND1', 1, 1) ],
+        'His41(+)-Cys145(-)-His163(+)' : [ ('HIS:41:.*:.*:.*', 'ND1', 1, 1), ('HIS:41:.*:.*:.*', 'NE2', 1, 1), ('CYS:145:.*:.*:.*', 'SG', -1, 0), ('HIS:163:.*:.*:.*', 'NE2', 1, 1), ('HIS:163:.*:.*:.*', 'ND1', 1, 1) ],
+    }
+    # Filter variants if desired
+    if receptor_protonation_states is not None:
+        variants = { name : data for (name, data) in variants.items() if name in receptor_protonation_states }
+
+    def generate_variant(design_unit, modifications):
+        from openeye import oedocking
+        protein = oechem.OEGraphMol()
+        design_unit.GetProtein(protein)
+
+        protonate_opts = opts.GetPrepOptions().GetProtonateOptions();
+        place_hydrogens_opts = protonate_opts.GetPlaceHydrogensOptions()
+
+        for (residue_match, atom_name, formal_charge, implicit_hydrogen_count) in modifications:
+            pred = oechem.OEAtomMatchResidue(residue_match)
+            place_hydrogens_opts.SetBypassPredicate(pred)
+            for atom in protein.GetAtoms(pred):
+                if oechem.OEGetPDBAtomIndex(atom) == getattr(oechem, f'OEPDBAtomName_{atom_name}'):
+                    #print('Modifying CYS 145 SG')
+                    oechem.OESuppressHydrogens(atom)
+                    atom.SetFormalCharge(formal_charge)
+                    atom.SetImplicitHCount(implicit_hydrogen_count)
+        # Update the design unit with the modified formal charge for CYS 145 SG
+        oechem.OEUpdateDesignUnit(design_unit, protein, oechem.OEDesignUnitComponents_Protein)
+
+        # Don't flip Gln189
+        pred = oechem.OEAtomMatchResidue(["GLN:189: :A"])
+        protonate_opts = opts.GetPrepOptions().GetProtonateOptions();
+        place_hydrogens_opts = protonate_opts.GetPlaceHydrogensOptions()
+        place_hydrogens_opts.SetNoFlipPredicate(pred)
```

Comment:
This is giving me downstream issues with OpenMM complaining about protonation states. For some PDBs (I can provide exmaples if needed) this set of `place_hydrogens_opts` produces wonky forms of histidine (e.g. all single bonds for some reason?). Using this block instead:
```
    pred = oechem.OEAtomMatchResidue(["GLN:189:.*:.*:.*"])
    protonate_opts = opts.GetPrepOptions().GetProtonateOptions();
    place_hydrogens_opts = protonate_opts.GetPlaceHydrogensOptions()
    place_hydrogens_opts.SetBypassPredicate(pred)
    place_hydrogens_opts.SetNoFlipPredicate(pred)
    protonate_opts = oespruce.OEProtonateDesignUnitOptions(place_hydrogens_opts) # ?
    opts.GetPrepOptions().SetProtonateOptions(protonate_opts) # ?
```
Fixes _most_ of the issues. Perhaps @kaminow or @apayne97 can comment on the best practice here?